Add docs for localization 8 (#30021)
dotnet/AspNetCore.Docs: aspnetcore/fundamentals/localization/select-language-culture.md

author: rick-anderson
description: Learn how to select a language and culture when localizing content into different languages and cultures in an ASP.NET Core app.
ms.author: riande

> monikerRange: '>= aspnetcore-8.0'

Reviewer: ```suggestion
monikerRange: '>= aspnetcore-5.0'
```

author: rick-anderson
description: Learn how to select a language and culture when localizing content into different languages and cultures in an ASP.NET Core app.
ms.author: riande

> 

Reviewer: monikerRange: '>= aspnetcore-5.0'

> 

Reviewer: ```suggestion
monikerRange: '>= aspnetcore-5.0'
```

As previously mentioned, add a custom provider via <xref:Microsoft.AspNetCore.Builder.RequestLocalizationOptionsExtensions.AddInitialRequestCultureProvider%2A> which sets the order to `0`, so this provider takes the precedence over the others.

:::moniker-end

> :::moniker range="> aspnetcore-8.0"

Reviewer: ```suggestion
:::moniker range=">= aspnetcore-8.0"
```

## User override culture

> In previous versions, there was no way a user could override the culture settings. The culture settings were always set by the system culture settings. In ASP.NET Core 8.0, the [RequestLocalizationOptions.CultureInfoUseUserOverride](xref:Microsoft.AspNetCore.Builder.RequestLocalizationOptions.CultureInfoUseUserOverride) property allows the user to override the culture settings:

Reviewer 2: > In previous versions, there was no way a user could override the culture settings.

I don't think this adds any value to most readers of this doc. Most features didn't exist in some previous versions. Mentioning the version it was introduced in is okay. It might be helpful in case people are cross targeting or something, but we don't need more than that.

I also think this is overstating what was possible in previous version. Before it was impossible to *prevent* the "user" of the system from overriding the web apps behavior by changing Window's default culture settings. At least not with `UseRequestLocalization`.

However, assuming the Windows system culture hasn't been customized, or you're running on Linux, this setting has no effect. That's why we've gone this long before someone asked for this feature. I would suggest linking to the docs for `CultureInfo.UseUserOverride`. That's why this property has that name just without the `.`. It's a very nuanced behavior, and we didn't want to imply it does more than it really does.

```suggestion
Starting in ASP.NET Core 8.0, the [RequestLocalizationOptions.CultureInfoUseUserOverride](xref:Microsoft.AspNetCore.Builder.RequestLocalizationOptions.CultureInfoUseUserOverride) property allows the application to decide whether or not to use non-default Windows settings for the <xref:System.Globalization.CultureInfo>'s <xref:System.Globalization.CultureInfo.DateTimeFormat> and <xref:System.Globalization.CultureInfo.numberformat>. This has no impact on Linux. This directly corresponds to <xref:System.Globalization.CultureInfo.UseUserOverride>.
```

```csharp
    app.UseRequestLocalization(options =>
    {
        options.CultureInfoUseUserOverride = false;
    });

> ```

Reviewer: @halter73  can you approve this to go into [What's new in ASP.NET Core 8.0](https://learn.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-8.0?view=aspnetcore-7.0)
Once you approve I'll have @hishamco  copy/paste this into What's new and do a pr there. I'll clean this up a bit for this document.

> Starting in ASP.NET Core 8.0, the [RequestLocalizationOptions.CultureInfoUseUserOverride](xref:Microsoft.AspNetCore.Builder.RequestLocalizationOptions.CultureInfoUseUserOverride) property allows the application to decide whether or not to use non-default Windows settings for the <xref:System.Globalization.CultureInfo>'s <xref:System.Globalization.CultureInfo.DateTimeFormat> and <xref:System.Globalization.CultureInfo.numberformat>. This has no impact on Linux. This directly corresponds to <xref:System.Globalization.CultureInfo.UseUserOverride>.

Reviewer 2: ```suggestion
Starting in ASP.NET Core 8.0, the [RequestLocalizationOptions.CultureInfoUseUserOverride](xref:Microsoft.AspNetCore.Builder.RequestLocalizationOptions.CultureInfoUseUserOverride) property allows the application to decide whether or not to use non-default Windows settings for the <xref:System.Globalization.CultureInfo> <xref:System.Globalization.CultureInfo.DateTimeFormat> and <xref:System.Globalization.CultureInfo.NumberFormat> properties. This has no impact on Linux. This directly corresponds to <xref:System.Globalization.CultureInfo.UseUserOverride>.
```
